elasticai/creator/io_table.py

```diff
@@ -0,0 +1,72 @@
+from typing import Union, Tuple, List, Dict
+
+
+import numpy as np
+
+
+class IOTable:
+    def __init__(self, inputs: np.ndarray, outputs: np.ndarray):
+        self.inputs = inputs
+        self.outputs = outputs
+        pass
+
+    @property
+    def tables(self):
+        return self.inputs, self.outputs
+
+    def __getitem__(self, item):
+        return (self.inputs[item], self.outputs[item])
+
+    def get_table_as_dict(self) -> Dict:
+        """given a list or single input,output table pair will return a list of dictionaries for each io pair. Said tables will be flatenned.
+        Args:
+            tables: io tables in the format
+        Returns:
+          dict: A list of dictionaries for the io_tables both ends will be flattened and transformed into tuples to facilitate iterating.
+        """
+
+        dic_table = {}
+        n = self.tables
+        for input, output in self:
+            dic_table[tuple(input.flatten().tolist())] = tuple(
+                output.flatten().tolist()
+            )
+        return dic_table
+
+
+def group_tables(table: IOTable, groups: int) -> List[IOTable]:
```

Comment:
cool! The next thing we need is a way to express which outputs comin from layer l_i flow into which inputs of layer l_(i+1)